vllm/executor/executor_base.py

```diff
@@ -110,6 +111,35 @@ def __del__(self):
 
 class ExecutorAsyncBase(ExecutorBase):
 
+    def __init__(
+        self,
+        model_config: ModelConfig,
+        cache_config: CacheConfig,
+        parallel_config: ParallelConfig,
+        scheduler_config: SchedulerConfig,
+        device_config: DeviceConfig,
+        load_config: LoadConfig,
+        lora_config: Optional[LoRAConfig],
+        vision_language_config: Optional[VisionLanguageConfig],
+        speculative_config: Optional[SpeculativeConfig],
+    ) -> None:
+        self.lock = asyncio.Lock()
+        self.pp_locks = [
+            asyncio.Lock()
+            for _ in range(parallel_config.pipeline_parallel_size)
+        ]  # This locks each pipeline parallel stage so multiple
+        # virtual engines can't execute on the same stage at the same time
+        self.pp_semaphores = [
+            asyncio.Semaphore(2)
+            for _ in range(parallel_config.pipeline_parallel_size)
+        ]  # This semaphore is used to limit the number of pipeline
```

Comment:
What happens if you don't have these semaphores?

Also, style nit, but I think it's a bit clearer if you put the comments before the variable declaration instead of after.

Reply:
Hmm actually is this used?

Reply:
Nope, I'll gonna get rid of it

Reply:
Done